core/src/test/java/com/graphhopper/reader/osm/OSMReaderTest.java

```diff
@@ -579,6 +580,41 @@ public void testTurnRestrictions() {
         assertTrue(tcStorage.get(bikeTCEnc, edge10_11, n11, edge11_14) > 0);
     }
 
+    @Test
+    void testViaWayTurnRestrictions() {
+        ReaderRelation rel = new ReaderRelation(1L);
+
+        rel.setTag("restriction", "no_u_turn");
+        rel.add(new ReaderRelation.Member(ReaderElement.Type.WAY, 1L, "from"));
+        rel.add(new ReaderRelation.Member(ReaderElement.Type.WAY, 2L, "via"));
+        rel.add(new ReaderRelation.Member(ReaderElement.Type.WAY, 3L, "to"));
+
+        List<OSMTurnRestriction> osmRel = OSMReader.createTurnRestrictions(rel);
+        assertEquals(osmRel.get(0).getViaType(), OSMTurnRestriction.ViaType.WAY);
+        assertEquals(osmRel.get(0).getViaOSMIds().get(0), 2);
+    }
+
+    @Test
+    void testMultiViaWayTurnRestrictions() {
+        ReaderRelation rel = new ReaderRelation(1L);
+
+        rel.setTag("restriction", "no_u_turn");
+        rel.add(new ReaderRelation.Member(ReaderElement.Type.WAY, 1L, "from"));
+        rel.add(new ReaderRelation.Member(ReaderElement.Type.WAY, 2L, "via"));
+        rel.add(new ReaderRelation.Member(ReaderElement.Type.WAY, 3L, "via"));
+        rel.add(new ReaderRelation.Member(ReaderElement.Type.WAY, 4L, "via"));
+        rel.add(new ReaderRelation.Member(ReaderElement.Type.WAY, 5L, "to"));
+
+        ArrayList<Long> viaOSMIds = new ArrayList<>();
+        viaOSMIds.add(2L);
+        viaOSMIds.add(3L);
+        viaOSMIds.add(4L);
+
+        List<OSMTurnRestriction> osmRel = OSMReader.createTurnRestrictions(rel);
+        assertEquals(osmRel.get(0).getViaType(), OSMTurnRestriction.ViaType.MULTI_WAY);
```

Comment:
JUnit fun fact: The signature of `assertEquals` is `assertEquals(expected, actual)` not `assertEquals(actual, expected)`.

Reply:
No they just swapped the position of the message parameter (from front to back)'